data/src/test/java/org/apache/iceberg/io/TestWriterMetrics.java

```diff
@@ -98,9 +100,9 @@ public TestWriterMetrics(FileFormat fileFormat) {
 
   protected abstract T toGenericRow(int value, int repeated);
 
-  @Before
+  @BeforeEach
   public void setupTable() throws Exception {
-    File tableDir = temp.newFolder();
+    File tableDir = new File(tempDir, "table");
```

Comment:
this should use `Files.createTempDirectory(temp, "junit").toFile()`

Reply:
done